src/one_time/core.clj

```diff
@@ -4,6 +4,16 @@
             [one-time.totp  :as totp]
             [one-time.hotp  :as hotp]))
 
+(defn ^:private safe=
+  "Constant-time comparison (for numbers with equal base 10 length) for numeric
+  types, via bytewise string equality of the decimal representation."
+  [x y]
+  (let [x' (.getBytes (str x))
+        y' (.getBytes (str y))]
+    (and
+      (= (count x') (count y'))
```

Comment:
This count introduces a timing attack itself. It'll short-circuit and prevent the next line from executing. Providing a theoretical attacker with the information of what the right length is.

Reply:
Thanks for mentioning. We briefly touched on the subject. https://github.com/suvash/one-time/pull/11#discussion_r549831046 Given the specific context the library exists in, the length of tokens is already predetermined.